cmis_web_proxy/controllers/cmis.py

```diff
@@ -0,0 +1,433 @@
+            # the operation is on the CMIS content linked to the Odoo model
+            # instance
+            return True
+        cmis_object = repo.getObject(request_cmis_objectid)
+        # We can't use a CMIS query to check if a node is in the expected
+        # tree since the indexation is asynchronous. In place of a simple
+        # query we check if one of the paths of the node linked to the Odoo
+        # content instance is in one of the node paths of the requested
+        # cmis_object
+        child_paths = cmis_object.getPaths()
+        parent_paths = repo.getObject(token_cmis_objectid).getPaths()
+        for p in parent_paths:
+            for cp in child_paths:
+                if p in cp:
+                    return True
+        _logger.info("%s is not a child of %s", request_cmis_objectid,
+                     token_cmis_objectid)
+        return False
+
+    def _check_content_action_access(self, cmis_path, proxy_info, params,
+                                     model_inst):
+        """Check that the User has de required Permissioon on the Odoo model
+        instance to di the expected CMIS action
+        """
+        cmisaction = params.get('cmisaction')
+        if not cmisaction:
+            return True
+        operation = CMSI_ACTIONS_OPERATION_MAP.get(cmisaction)
+        if not operation:
+            _logger.info("CMIS action %s not supported", cmisaction)
+            return False
+        if not self._check_access_operation(model_inst, operation):
+            _logger.info("User don't have the access right for operation %s "
+                         "on %s to execute the CMIS action %s", operation,
+                         model_inst.name_get()[0][1], cmisaction)
+            return False
+        return True
+
+    def _check_access(self, cmis_path, proxy_info, params):
+        """This method check that the user can access to the requested CMIS
+        content.
+
+        Security checks applied  when the proxy mode is activated,:
+
+        1. Requests from the client must provide a token (in the header or
+           as param of the request).
+           If no security token is provided in this case, the access is denied.
+
+        2. The Odoo object referenced by the token (the token is build as
+           'model.name' + '_' + 'instance_id') must exist.
+
+        3. The user must have read access to the object referenced by the token
+
+        4. If a cmis_path or object_id is provided by the request, the
+           referenced CMIS content must be child of or the node referenced by
+           the Odoo object from the token (or equal)
+
+        5. If a cmisaction is provided by the request, a check is done to
+           ensure that the user has the required privileges in Odoo
+        """
+        # check token conformity
+        token = self._check_provided_token(cmis_path, proxy_info, params)
+        if not token:
+            raise AccessError("Bad request")
```

Comment:
From pylint: `String parameter of raise "AccessError" requires translation. Use _('Bad request')`